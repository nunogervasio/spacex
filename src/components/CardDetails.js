import React from "react";
import PropTypes from "prop-types";
import youtubeImg from "../img/youtube-1.svg";
import wikiImg from "../img/wiki-1.svg";
const CardDetails = ({
  patch_url,
  number,
  name,
  date,
  site,
  rocketName,
  rocketType,
  details,
  youtube_url,
  wiki_url
}) => {
  return (
    <div className="main-card-details">
      <div className="card-img-details">
        <img src={patch_url} className="patch" alt="mission patch" />
      </div>

      <div className="card-details-details">
        <p>
          <span className="flight-name">{name}</span>
          <br />
          <span className="card-location">Flight Number: </span> {number}
          <br />
          <span className="card-date-time">Date:</span> {date}
          <br />
          <span className="card-location">Location:</span> {site}
          <br />
          <span className="card-location">Rocket Name:</span> {rocketName}
          <br />
          <span className="card-location">Rocket Type:</span> {rocketType}
          <br />
          {/****** TODO: return "no details available" when there are no details 
          this also call a proptype error when details returns null*/}
          <span className="card-location">Mission Details: </span> {details}
          <br />
        </p>

        <div className="links">
          <div className="card-location clearfix">
            <a href={youtube_url} target="_blank" rel="noopener noreferrer">
              <img src={youtubeImg} alt="youtube icon" className="float-left" />
              <span className="youtube">YouTube Video</span>
            </a>
          </div>
          <div className="card-location clearfix">
            <a href={wiki_url} target="_blank" rel="noopener noreferrer">
              <img src={wikiImg} alt="wiki icon" className="float-left" />
              <span className="wiki">Wikipedia Page</span>
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CardDetails;

CardDetails.propTypes = {
  patch_url: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketType: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  youtube_url: PropTypes.string.isRequired,
  wiki_url: PropTypes.string.isRequired
};
