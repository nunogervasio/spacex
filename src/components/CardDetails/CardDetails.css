.main-card-details {
  border-top: solid black 2px;
  border-bottom: solid black 2px;
  padding-top: 16px;
  margin: 0.3em;
  display: flex;
  flex-direction: column;
}
.card-img-details {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.links {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-bottom: 1em;
  padding: 1em;
}
.card-details-details {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 1em;
}

.card-location img {
  max-height: 20px;
  max-width: 20px;
}
/* Clearfix (clear floats) */
.clearfix::after {
  content: "";
  clear: both;
  display: table;
}
/****** floated columns */
.youtube,
.wiki {
  float: right;
  margin-left: 5px;
}

.float-left {
  width: 25%;
  float: left;
}
@media (min-width: 769px) {
  .main-card-details {
    flex-direction: row;
  }
  .card-img-details {
    align-self: center;
  }
}
