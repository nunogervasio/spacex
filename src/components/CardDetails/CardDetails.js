import React from "react";
import PropTypes from "prop-types";
import youtubeImg from "../../img/youtube-1.svg";
import wikiImg from "../../img/wiki-1.svg";
import Patch from "../Patch/Patch";
import Links from "../Links/Links";
import MissionDetails from "../MissionDetails/MissionDetails";
import styles from "./CardDetails.module.css";

const CardDetails = ({
  patch_url,
  number,
  name,
  date,
  site,
  rocketName,
  rocketType,
  details,
  youtube_url,
  wiki_url
}) => {
  return (
    <div className={styles.card}>
      <Patch patch_url={patch_url} />
      <div className={styles.details}>
        <MissionDetails
          name={name}
          number={number}
          date={date}
          site={site}
          rocketName={rocketName}
          rocketType={rocketType}
          details={details}
        />
        <Links
          youtube_url={youtube_url}
          youtubeImg={youtubeImg}
          wiki_url={wiki_url}
          wikiImg={wikiImg}
        />
      </div>
    </div>
  );
};

export default CardDetails;

CardDetails.propTypes = {
  patch_url: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketType: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  youtube_url: PropTypes.string.isRequired,
  wiki_url: PropTypes.string.isRequired
};
