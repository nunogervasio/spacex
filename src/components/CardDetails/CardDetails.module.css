.card {
  border-top: solid black 2px;
  border-bottom: solid black 2px;
  padding-top: 16px;
  margin: 0.3em;
  display: flex;
  flex-direction: column;
}
.details {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 1em;
  padding: 0 0.3em;
}
@media (min-width: 769px) {
  .card {
    flex-direction: row;
  }
}
