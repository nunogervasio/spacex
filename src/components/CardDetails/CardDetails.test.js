import React from "react";
import { shallow } from "enzyme";
import CardDetails from "./CardDetails";

describe("Footer", () => {
  const props = {
    patch_url: "",
    number: "",
    name: "",
    date: "",
    site: "",
    rocketName: "",
    rocketType: "",
    details: "",
    youtube_url: "",
    wiki_url: ""
  };
  const cardDetails = shallow(<CardDetails {...props} />);
  it("renders properly", () => {
    expect(cardDetails).toMatchSnapshot();
  });
});
