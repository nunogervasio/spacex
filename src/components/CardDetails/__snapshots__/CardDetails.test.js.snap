// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDetails
    date=""
    details=""
    name=""
    number=""
    patch_url=""
    rocketName=""
    rocketType=""
    site=""
    wiki_url=""
    youtube_url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Patch
          patch_url=""
        />,
        <div
          className="details"
        >
          <MissionDetails
            date=""
            details=""
            name=""
            number=""
            rocketName=""
            rocketType=""
            site=""
          />
          <Links
            wikiImg="wiki-1.svg"
            wiki_url=""
            youtubeImg="youtube-1.svg"
            youtube_url=""
          />
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "patch_url": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MissionDetails
              date=""
              details=""
              name=""
              number=""
              rocketName=""
              rocketType=""
              site=""
            />,
            <Links
              wikiImg="wiki-1.svg"
              wiki_url=""
              youtubeImg="youtube-1.svg"
              youtube_url=""
            />,
          ],
          "className": "details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "date": "",
              "details": "",
              "name": "",
              "number": "",
              "rocketName": "",
              "rocketType": "",
              "site": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "wikiImg": "wiki-1.svg",
              "wiki_url": "",
              "youtubeImg": "youtube-1.svg",
              "youtube_url": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Patch
            patch_url=""
          />,
          <div
            className="details"
          >
            <MissionDetails
              date=""
              details=""
              name=""
              number=""
              rocketName=""
              rocketType=""
              site=""
            />
            <Links
              wikiImg="wiki-1.svg"
              wiki_url=""
              youtubeImg="youtube-1.svg"
              youtube_url=""
            />
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "patch_url": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MissionDetails
                date=""
                details=""
                name=""
                number=""
                rocketName=""
                rocketType=""
                site=""
              />,
              <Links
                wikiImg="wiki-1.svg"
                wiki_url=""
                youtubeImg="youtube-1.svg"
                youtube_url=""
              />,
            ],
            "className": "details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "date": "",
                "details": "",
                "name": "",
                "number": "",
                "rocketName": "",
                "rocketType": "",
                "site": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "wikiImg": "wiki-1.svg",
                "wiki_url": "",
                "youtubeImg": "youtube-1.svg",
                "youtube_url": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
